### DISCUSSION TITLE
question for testing new buttons

### DISCUSSION THREAD

**Discussion by angetheo-auth** [sButtons/sbuttons#1589] [Q&A] [answered]
Hello, 
I want to contribute to your project for an assignment at my university by implementing a simple mute button. My question is where do you suggest I test my button while Im making it. Im quite new at contributing to open source software and I want to make sure I follow the project's workflow as best as I can.
Also Im assuming that this type of button goes in the icon category? Please confirm
Thank you

**Comment by shahednasser** [comment, marked as answer]
Hello @angetheo-auth and thank you for your interest!

First, make sure you follow the [contributing guidelines](https://github.com/sButtons/sbuttons/blob/e653b5541fe2b2e100c325226828188d6469cbee/CONTRIBUTING.md) to understand how contributing works. You also need to submit an issue with a button idea first before working on it.

At the moment we're not accepting icon buttons unless the button has a special look rather than just a button with an icon.

After your button idea gets approved and you create the button in the designated `less` file (please refer to the contributing guidelines), you should compile the change by running:

```
npm run build
```

Then you need to add the button in `website/src/data/buttons.json` as explained in the contributing guidelines.

After that, you should start the server under the directory `website`, and you should see your button there!
